### DISCUSSION TITLE
Why using the Controller and FormProvider instead of using the Ref and register?

### DISCUSSION THREAD

**Discussion by matheusrocha89** [shadcn-ui/ui#2679] [Q&A]
Checking the documentation of `react-hook-form` and the examples here on the `shadcn-ui`, the Form component uses the `FormProvider` from `react-hook-form` and the Controlled components example. But most of the components of the Shadcn-ui forward the ref.
Was there any reason not to use the examples using the `register` from `react-hook-form` and choose the controlled components approach?

**Comment by mahendraAtMave** [comment]
same question
